src/pizzeria/Pizza.java

```diff
@@ -22,32 +22,37 @@ public Pizza(String name, PizzaType pizzaType) {
     }
 
     public Pizza(String name, PizzaType pizzaType, Ingredient[] ingredients) {
+        if (name.length() < 4 || name.length() > 20) {
+            name = "Customer_name_";
+        }
         this.name = name;
         this.pizzaType = pizzaType;
         if (ingredients.length > MAX_ALLOWED_INGREDIENTS_SIZE) {
-            System.out.println("Ingredients more then allowed " + MAX_ALLOWED_INGREDIENTS_SIZE);
+            System.out.println("Ingredient are more than allowed");
             this.ingredients = new Ingredient[MAX_ALLOWED_INGREDIENTS_SIZE];
         } else {
             this.ingredients = ingredients;
         }
     }
 
-    public double calculatePrice() {
+    public double calculatePrice () {
         double price = 0;
         price = price + pizzaType.getPrice();
-
         for (Ingredient ingredient : ingredients) {
             price += ingredient.getPrice();
         }
-
         return price;
     }
 
-    public void addIngredients(Ingredient ingredient) {
+    public void addIngredients (Ingredient ingredients) {
```

Comment:
Change this method name, because you add **ONE** ingredient -> **addIngerdient(Ingredient ingredient)**

Reply:
done